picoquic/packet.c

```diff
@@ -93,6 +92,23 @@ int picoquic_parse_packet_header(
                 }
                 else
                 {
+                    /* Is the context found by using the `addr_from`? */
+                    char context_by_addr = 0;
+                    /* Retrieve the connection context */
+                    if (*pcnx == NULL)
+                    {
+                        *pcnx = picoquic_cnx_by_id(quic, ph->cnx_id);
+
+                        /* TODO: something for the case of client initial, e.g. source IP + initial CNX_ID */
+                        if (*pcnx == NULL)
+                        {
+                            *pcnx = picoquic_cnx_by_net(quic, addr_from);
+                            context_by_addr = 1;
+                        }
+                    }
+
+                    char is_client = *pcnx && (*pcnx)->client_mode;
```

Comment:
I always try to place declarations at the beginning of the block. Is it because I am too old school? That used to be mandated by the old C compilers...

Reply:
Yeah the old days ^^ I will try to do so in future commits :)